bgpd/bgp_nhc.c

```diff
@@ -87,8 +87,11 @@ uint64_t bgp_nhc_nnhn_count(struct bgp_nhc *nhc)
 
 	for (tlv = nhc->tlvs; tlv; tlv = tlv->next) {
 		if (tlv->code == BGP_ATTR_NHC_TLV_NNHN) {
-			/* BGP Identifier is always 4-bytes (yet...) */
-			count = tlv->length / IPV4_MAX_BYTELEN;
+			/* BGP Identifier is always 4-bytes (yet...).
+			 * -1 is to exclude the next-hop BGP ID.
+			 * We care only about Next-next hops here.
+			 */
+			count = (tlv->length / IPV4_MAX_BYTELEN) - 1;
```

Comment:
Can this over (under) flow? boundary check?

Reply:
Added a guard, thanks.